### PR TITLE
adding more coverage to the php functions

### PR DESCRIPTION
**Pull request by mayank**
## Description of new Instant Answer, or changes
php: udpated the coverage to cover more functions

* covered functions from classes (~6000 are remaining)
* double documentation with classname and without classname both

## Related Issues and Discussions
Improvement: #616

## People to notify
@VitorVRS @gautamkrishnar

## Testing & Review
To be completed by Language Leader (or DDG Staff) when reviewing Pull Request
* Tested by @gautamkrishnar : Not triggering. Found duplicates.

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page**
- [x] Instant Answer page is correctly filled out and contains:
    - One topic for the Search Space Language (Java, Python, Scala, Ruby, etc.)
    - One topic from: Reference, Help, Libraries, Tools
        - Documentation Fatheads are considered "Reference"
    - Description, Data source, and 2+ example queries
    - Perl Module (e.g. "DDG::Fathead::PerlDoc" -- we only need a name, not an actual file)
    - Source Name (for "More at <source_name>" link)
    - Source Domain (must contain http:// or https:// -- can be the same as Data Source)
    - Source Info (used as Subtitle for each Article -- usually matches the IA Name)
    - 'Skip Abstract' is checked off
    - Source ID (ping @moollaza to assign one, once Fathead is ready for Beta deploy)
- [ ] Fathead Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Code**
- [x] Uniformly indented, well commented
- [x] Fetch.sh and Parse.sh run without errors
- [x] Output contains no blank lines, or multi-line entries

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/php

### CODE REVIEW

**Comment by daxtheduck** [comment]
@mayank Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by gautamkrishnar** [comment]
Duplicates:
```
2930 Duplicates found
```
Test results:
```bash

#   Failed test 'Testing article coverage'
#   at t/validate_fathead.t line 48.
# NO TRIGGER WORDS FOUND. Will not be able to validate categories
#
# ...Finished
#
# Looks like you failed 3 tests of 8.
[ERROR]  Tests failed! See output above for details
```

**Comment by mayank** [comment]
@gautamkrishnar sorry, merged some wrong code.. will update this in a while.
Can we add something like travis CI which gives a pre-build status?

**Comment by gautamkrishnar** [comment]
@mayank yes we could add something to travis CI So that it automaticially build the changed IA,
What's your opinion @duckduckgo/community-leaders, @moollaza, @tagawa

**Comment by mayank** [comment]
@gautamkrishnar done!

- removed ignored files list, now parsing whole documentation
- parsing only code having functions

**Comment by gautamkrishnar** [comment]
Testing... 💻 🖱

**Comment by mayank** [comment]
Hi @gautamkrishnar, build is failing because all the keywords are not covered some of which are classes, reserved keywords, I only covered the functions.

Also it is same as on master.
Results of `duckduckgo/zeroclickinfo-fathead:master`
```

#   Failed test 'Testing article coverage'
#   at t/validate_fathead.t line 48.
# NO TRIGGER WORDS FOUND. Will not be able to validate categories
#
# ...Finished
#
# Looks like you failed 2 tests of 8.
t/validate_fathead.t .. Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/8 subtests
        (less 3 skipped subtests: 3 okay)

Test Summary Report
-------------------
t/validate_fathead.t (Wstat: 512 Tests: 8 Failed: 2)
  Failed tests:  4, 7
  Non-zero exit status: 2
Files=1, Tests=8,  7 wallclock secs ( 0.12 usr  0.09 sys +  2.04 cusr  0.18 csys =  2.43 CPU)
Result: FAIL
[ERROR]  Tests failed! See output above for details
```

**Comment by gautamkrishnar** [comment]
Thanks @mayank please let me know when you have finished this

**Comment by mayank** [comment]
@gautamkrishnar finished! seems working fine.

image

**Comment by gautamkrishnar** [comment]
Still it is not triggering for me in codio.

**Comment by gautamkrishnar** [comment]
Not triggering for "php array merge" query

**Comment by gautamkrishnar** [comment]
I thing it shouldn't be triggered without the php keyword at start end. 
Its now triggering for `gettotalsize`?  gettotalsize is also a function in some other programming language...
@moollaza @mintsoft what's your opinion?

**Comment by mayank** [comment]
I think `duckpan` doesn't fully imitate production env. This will work fine on production as it is not able to select Topic from the json.

**Comment by mayank** [comment]
@gautamkrishnar it seems that it works same as for other fatheads. Can we deploy this on beta?

**Comment by gautamkrishnar** [comment]
Let's wait for @moolaza...
Cc: @MariagraziaAlastra

**Comment by gautamkrishnar** [comment]
Working fine on beta
tmp_6985-screenshot_20170109-201622-1577628727

**Comment by mayank** [comment]
@gautamkrishnar thanks 😄

File: lib/fathead/php/parse.js [hunk 1]
```diff
@@ -92,9 +92,10 @@
     function parserFunction(className, key, fileName) {
         if(debug) console.log('parsing: '+chalk.blue(className)+' '+chalk.cyan(key)+' '+chalk.green(fileName))
         
-	// checking if function already exists, if not also add name without class
-	if(className == 'function') {
+	    // checking if function already exists, if not also add name without class
+	    if(className == 'function') {
```
Comment (moollaza) [inline lib/fathead/php/parse.js]: nit: the indentation here is wrong...

**Comment by moollaza** [comment]
> I think duckpan doesn't fully imitate production env.

@mayank you are correct -- DuckPAN doesn't fully replicate the production environment. It's mostly for testing the visual display of your Fathead

**Comment by moollaza** [comment]
@mayank Thanks! This LGTM :+1:

**Comment by mayank** [comment]
@moollaza how to test for category pages? DuckPAN does not show category pages.

**Comment by moollaza** [comment]
> @moollaza how to test for category pages? DuckPAN does not show category pages.

@mayank they can be tested when deployed to Beta. Any Fathead that belongs to a category should contain a link to it

**Pull request merged**